File: DoConnect/src/main/java/com/wipro/doconnect/service/AdminService.java
package com.wipro.doconnect.service;

import java.util.List;
import com.wipro.doconnect.dto.ResponseDTO;
import com.wipro.doconnect.entity.Admin;
import com.wipro.doconnect.entity.Answer;
import com.wipro.doconnect.entity.Question;
import com.wipro.doconnect.entity.User;
/*@Author:Saiteja Velamala
 * Modified Date: 27-08-2022
 * Description:Admin Service
 */
public interface AdminService
{
	public Admin adminLogin(String email, String password);
	public String adminLogout(Long adminId);
	public Admin adminRegister(Admin admin);
	
	public Question approveQuestion(Long questionId);
	public Answer approveAnswer(Long answerId);
	
	public List<Question> getUnApprovedQuestions();
	public List<Answer> getUnApprovedAnswers();
	
	public ResponseDTO deleteQuestion(Long questionId);
	public ResponseDTO deleteAnswer(Long answerId);
	
	public User getUser(String email);
	public List<User> getAllUser();
}

File: DoConnectUI/src/app/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Answer } from './answer';
import { Question } from './question';
import { User } from './user';
import { Admin } from './admin';

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http:HttpClient) {

   }

   admin= new Admin()

   url:string="http://localhost:9090/admin"

   sendAdminData(admin:Admin){
    console.log("admin data coming from "+admin.id)
    this.admin.id=admin.id
    this.admin.email=admin.email
    this.admin.name=admin.name
   }

   giveAdminData(){
    return this.admin
   }

   adminLogin(email:string,password:string):Observable<Admin>{
    return this.http.get<Admin>(this.url+"/login/"+email+"/"+password)
   }

   adminLogout(adminId:number):Observable<string>{
    return this.http.get<string>(this.url+"/logout/"+adminId)
   }

   adminRegister(admin:Admin):Observable<Admin>{
    return this.http.post<Admin>(this.url+"/register",admin)
   }

   getUnApprovedQuestions():Observable<Question[]>{
     return this.http.get<Question[]>("http://localhost:9090/admin"+"/getUnApprovedQuestions")
   }
   getUnApprovedAnswers():Observable<Answer[]>{
     return this.http.get<Answer[]>(this.url+"/getUnApprovedAnswers")
   }

   question={
  }
   approveQuestion(questionId:number):Observable<Question>{
     return this.http.put<Question>("http://localhost:9090/admin/approveQuestion/"+questionId,this.question)

   }

   approveAnswer(answerId:number):Observable<Answer>{
     return this.http.put<Answer>(this.url+"/approveAnswer/"+answerId,Answer)
   }
   deleteQuestion(questionId:number){
     return this.http.delete(this.url+"/deleteQuestion/"+questionId)

   }
   deleteAnswer(answerId:number){
     return this.http.delete(this.url+"/deleteAnswer/"+answerId)

   }
   getUser(email:string):Observable<User> {
     return this.http.get<User>(this.url+"/getUser/"+email);
   }

   getAllUsers():Observable<User[]>{
    return this.http.get<User[]>(this.url+"/getAllUsers")
   }

}
